src/api/app/views/webui2/webui/request/_request_history.html.haml

```diff
@@ -0,0 +1,47 @@
+%ul.list-unstyled
+  %li.media
+    - user = User.find_by_login(bs_request.creator)
+    = image_tag_for(user, size: 32, custom_class: 'mr-3', width: nil)
+    .media-body
+      %h5.mt-0.mb-1 #{bs_request.creator} (#{fuzzy_time(bs_request.created_at)})
+      created request
+      .media.mt-2
+        .media-body
+          - unless bs_request.description.blank?
+            .req-description-box
+              - if bs_request.description.length > 100
+                = simple_format(bs_request.description, class: 'collapsed-history', id: 'req-description')
+                = link_to('', '###', id: "show-full-req-description", class: 'collapsed-link', title: 'show more')
+              - else
+                = simple_format(bs_request.description)
+
+  - history.each_with_index do |element, index|
+    %li.media.mt-3
+      = image_tag_for(element.user, size: 32, custom_class: 'mr-3', width: nil)
+      .media-body
+        %h5.mt-0.mb-1 #{element.user.login} (#{fuzzy_time(element.created_at)})
+        - if element.class == HistoryElement::RequestSuperseded
+          = link_to(element.description, request_show_path(element.description_extension))
+        - else
+          #{element.description}
+        - unless element.comment.blank?
+          .media.mt-2
+            .media-body
+              .req-description-box
+                - if element.comment.length > 100
+                  = simple_format(element.comment, class: 'collapsed-history', id: 'req-description')
+                  = link_to('', '###', class: 'collapsed-link', id: 'show-full-req-description', title: 'show more')
+                - else
+                  = simple_format(element.comment)
+
+= content_for :ready_function do
+  :plain
+    $('.req-description-box').on('click', '#req-description', function() {
```

Comment:
why do you need the `#req-description` paramether?

Reply:
repeated code, you can move it to a method :wink:

Reply:
or partial..

Reply:
@krauselukas the pattern ``` @mycollection.each { |e| render partial: 'foo', locals {} ``` is one that makes the partial usage slow, specially in the history section which can become huge and there is no way to support pagination.... I would suggest you to use partials but to pass the collection to render i.e ```= render partial: 'foo', collection: @my_collection```

some reading material:
https://medium.com/@coorasse/partial-rendering-performance-in-rails-101fdfb6ffb9
https://api.rubyonrails.org/classes/ActionView/PartialRenderer.html

Reply:
I agree with @Ana06, this could into a function to avoid repeated code